internal/datautils/job_data.go

```diff
@@ -26,4 +26,7 @@ type StepRunData struct {
 
 	// overrides set from the playground
 	Overrides map[string]interface{} `json:"overrides"`
+
+	// errors in upstream steps (only used in on-failure step)
+	Errors map[string]string `json:"errors,omitempty"`
```

Comment:
Is this a reasonable way to go?

Reply:
I think so considering the comment -- `Errors` could be construed with internal errors from the engine, so perhaps `StepRunErrors` would make sense, but I don't have a strong opinion

Reply:
This sounds like a better name to me, I'll go with this - think it'll be easier to understand on the client side 👍